packages/react-composites/src/composites/ChatComposite/file-sharing/FileDownloadHandler.ts

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { CommunicationIdentifierKind } from '@azure/communication-common';
+import { FileMetadata } from './FileUpload';
+
+/**
+ * @beta
+ *
+ * A callback function for handling file downloads.
+ * The function needs to return a promise that resolves to a file download URL.
+ * If the promise is rejected, the {@link Error.message} will be used to display an error message to the user.
+ *
+ * Sample Usage:
+ * ```ts
+ * const fileDownloadHandler: FileDownloadHandler = async (userId, fileData) => {
+ *   if (isUnauthorizedUser(userId)) {
+ *     throw new Error('You don’t have permission to download this file.');
+ *   } else {
+ *     return fileData.url;
+ *   }
+ * }
+ *
+ * const App = () => (
+ *   <ChatComposite
+ *     ...
+ *     fileSharing={{
+ *       fileDownloadHandler: fileDownloadHandler
+ *     }}
+ *   />
+ * )
+ *
+ * ```
+ * @param userId - The {@link CommunicationIdentifierKind} ID of the user uploading the file.
+ * @param fileData - The {@link FileMetadata} containing file `url`, `extension` and `name`.
+ */
+export type FileDownloadHandler = (userId: CommunicationIdentifierKind, fileData: FileMetadata) => Promise<string>;
```

Comment:
Not a comment, just a question. Where will we implement the FileDownloadHandler or is Contoso going to handle that.

Reply:
Also, don't we need threadId as parameter for FileDownloadHandler or is it included in metadata.

Reply:
Contoso needs to implement this function.
We don't need the threadId since we are providing a userID already.
How do you think a threadId might be used here?

Reply:
Don't we need threadId to see which chat thread the message belongs to? User might be active in more than one threads.

Reply:
For a file download scenario, contoso doesn't need to know about the thread ID.
They need to know which file is being downloaded and by who.